mlflow/store/db_migrations/versions/0584bdc529eb_add_cascading_deletion_to_datasets_from_experiments.py

```diff
@@ -44,15 +68,8 @@ def upgrade():
                 ondelete="CASCADE",
             )
     else:
-        if dialect_name == "postgresql":
-            fk_constraint_name = f"{SqlDataset.__tablename__}_experiment_id_fkey"
-        elif dialect_name == "mysql":
-            fk_constraint_name = f"{SqlDataset.__tablename__}_ibfk_1"
-        elif dialect_name == "mssql":
-            # mssql fk constraint name at f5a4f2784254
-            fk_constraint_name = f"FK__{SqlDataset.__tablename__}__experi__6477ECF3"
-
-        op.drop_constraint(fk_constraint_name, SqlDataset.__tablename__, type_="foreignkey")
+        old_fk_constraint_name = get_datasets_experiment_fk_name()
```

Comment:
if we want to be safe, i can retain the old behavior (hardcode for all db flavors) and just attempt to query for MSSQL